src/components/cc-input-date/cc-input-date.js

```diff
@@ -0,0 +1,757 @@
+ */
+function getDateOrParseIso (dateOrString) {
+  if (isDateValid(dateOrString)) {
+    return dateOrString;
+  }
+  return ISO_DATE_PARSER.parse(dateOrString);
+}
+
+/**
+ * @return {InputDateValueStateEmpty}
+ */
+function dateStateEmpty () {
+  return {
+    state: 'empty',
+  };
+}
+
+/**
+ * @param {string} value
+ * @return {InputDateValueStateNaD}
+ */
+function dateStateNaD (value) {
+  return {
+    state: 'NaD',
+    value,
+  };
+}
+
+/**
+ * @param {Date} date
+ * @return {InputDateValueStateValid}
+ */
+function dateStateValid (date) {
+  return {
+    state: 'valid',
+    value: date.toISOString(),
+    date,
+  };
+}
+
+const VALID = { valid: true };
+
+/**
+ * @param {T} code
+ * @return {{valid: false, code: T}}
+ * @template T
+ */
+function invalid (code) {
+  return {
+    valid: false,
+    code,
+  };
+}
+
+/**
+ * @typedef {import('../../lib/date/date.types.js').Timezone} Timezone
+ * @typedef {import('./cc-input-date.types.js').InputDateValueState} InputDateValueState
+ */
+
+/**
+ * A custom date input.
+ *
+ * ## Technical details
+ *
+ * * Uses a native `<input>` with a type `string`
+ * * When an error slot is used, the input is decorated with a red border and a redish focus ring. You have to be aware that it uses the [`slotchange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event) event which doesn't fire if the children of a slotted node change.
+ *
+ * @cssdisplay inline-block
+ *
+ * @event {CustomEvent<string>} cc-input-date:input - Fires the `value` whenever the `value` changes.
+ * @event {CustomEvent} cc-input-date:requestimplicitsubmit - Fires when enter key is pressed.
+ *
+ * @cssprop {FontFamily} --cc-input-font-family - The font-family for the input content (defaults: `inherit`).
+ *
+ * @slot error - The error message to be displayed below the `<input>` element or below the help text. Please use a `<p>` tag.
+ * @slot help - The help message to be displayed right below the `<input>` element. Please use a `<p>` tag.
+ */
+export class CcInputDate extends LitElement {
+
+  static get properties () {
+    return {
+      disabled: { type: Boolean, reflect: true },
+      hiddenLabel: { type: Boolean, attribute: 'hidden-label' },
+      inline: { type: Boolean, reflect: true },
+      label: { type: String },
+      max: { type: String },
+      min: { type: String },
+      name: { type: String, reflect: true },
+      readonly: { type: Boolean, reflect: true },
+      required: { type: Boolean },
+      skeleton: { type: Boolean, reflect: true },
+      timezone: { type: String },
+      value: { type: String },
+      _valueState: { type: Object, state: true },
+      _hasError: { type: Boolean, state: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {boolean} Sets `disabled` attribute on inner native `<input>` element. */
+    this.disabled = false;
+
+    /** @type {boolean} Hides the label visually if `true`. */
+    this.hiddenLabel = false;
```

Comment:
If we want to design a `cc-date-range` component, maybe we would like to have only one label for the date range, and no labels for the two `cc-input-date`.

Maybe, this is a bad idea in terms of accessibility. What do you think?

Reply:
This is not the best for accessibility but I guess it would be acceptable if we specify the format within the `legend` of the `fieldset` grouping both these components :eyes:.

I guess we can allow it and do our best to avoid using it (test every design option before falling back to hiding labels).

Let's keep this because your point is definitely valid :+1:.